eng/pipelines/templates/stages/archetype-python-release.yml

```diff
@@ -271,3 +271,34 @@ stages:
               twine upload --repository $(DevFeedName) --config-file $(PYPIRC_PATH) $(Pipeline.Workspace)/${{parameters.ArtifactName}}/${{artifact.name}}/*-*a*.zip
               echo "Uploaded sdist to devops feed $(DevFeedName)"
             displayName: 'Publish ${{artifact.name}} alpha package'
+
+    - job: PublishDocsToNightlyBranch
+      condition: or(eq(variables['SetDevVersion'], 'true'), and(eq(variables['SetDevVersion'], ''), eq(variables['Build.Reason'],'Schedule'), eq(variables['System.TeamProject'], 'internal')))
+      pool:
+        name: azsdk-pool-mms-ubuntu-2004-general
+        vmImage: MMSUbuntu20.04
+      steps:
```

Comment:
These steps also belong up above: https://github.com/Azure/azure-sdk-for-python/pull/20809/files#diff-16ece2f30d68fa34c3ff0872cd2d53fe796f99709e70e0faf8cfbef9e5e9c3f0R146

This will enable publishing to update the `main` branch in the docs repo and move that process forward.

Reply:
Why docs main need this? Can we have metadata only in `daily` branch? docindex will update the ci-config in daily branch and public to Doc.Ms review daily branch. I did not see the necessity of having metadata in `main` branch of docs repo

Reply:
Chat offline. Will update main branch metadata on release pipeline.